staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
+}
+
+// BearerTokenSource is used to generate
+type BearerTokenSource interface {
+
+	// Token returns a bearer token to add to all requests sent by the REST client.
```

Comment:
We can define the priority here: https://github.com/kubernetes/kubernetes/pull/45817/files#diff-f5423923c5f61f08517f3eb0014e48eeR38